### PR TITLE
Update `setup.ilm.check_exists` documentation

### PR DESCRIPTION
**Pull request by belimawr**
## Proposed commit message

This commit updates the documentation from `setup.ilm.check_exists` in our reference configuration files to better explain how it works and align the wording with our public documentation.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

~~## Author's Checklist~~
~~## How to test this PR locally~~
## Related issue

- Closes https://github.com/elastic/beats/issues/33938

~~## Use cases~~
~~## Screenshots~~
~~## Logs~~

### CODE REVIEW

**Review by rdner — APPROVED** [review]

**Comment by pierrehilbert** [comment]
@narph could we please have someone to review here?

**Review by efd6 — APPROVED** [review]

**Comment by pierrehilbert** [comment]
Thanks @efd6 .
@belimawr this is now ready to be merged :-)

File: libbeat/idxmgmt/lifecycle/config.go [hunk 1]
```diff
@@ -33,9 +33,9 @@ type Config struct {
 	// used only for testing
 	policyRaw *Policy
 
-	// CheckExists can disable the check for an existing policy. Check required
-	// read_ilm privileges.  If check is disabled the policy will only be
-	// installed if Overwrite is enabled.
+	// CheckExists can disable the check for an existing policy. This check
+	// requires read_ilm privileges. If CheckExists is disabled the policy
```
Comment (ycombinator) [inline libbeat/idxmgmt/lifecycle/config.go:37]: Nit: additional clarity.

```suggestion
	// requires read_ilm privileges. If CheckExists is set to false, the policy
```

**Review by ycombinator — APPROVED** [review]
Left a nitpick.  LGTM - useful clarification about precedence for the user.

**Review by gizas — APPROVED** [review]

**Comment by pierrehilbert** [comment]
@nfritts / @andrewkroh could we please have someone to review here?

**Comment by belimawr** [comment]
I've enabled auto-merge, hopefully we can get this merged soon.

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**